udata/core/organization/api.py

```diff
@@ -255,8 +258,18 @@ class MembershipRequestAPI(API):
     @api.marshal_list_with(request_fields)
     def get(self, org):
         """List membership requests for a given organization"""
-        OrganizationPrivatePermission(org).test()
         args = requests_parser.parse_args()
+        if args["user"]:
+            if not current_user.is_authenticated or (
+                str(current_user.id) != args["user"]
+                and not OrganizationPrivatePermission(org).can()
+            ):
+                api.abort(
+                    403,
+                    "You can only access your own membership requests or the one of your organizations.",
+                )
+            return [r for r in org.requests if r.user.id == args["user"]]
```

Comment:
You do not check the `status` param?

Reply:
Indeed, after looking at the membership code, multiple requests are allowed for the same user and organization